app/services/joinable_builder.rb

```diff
@@ -0,0 +1,38 @@
+class JoinableBuilder < ApplicationService
```

Comment:
I'm always bigger on composition rather than inheritance for anything not a Controller. I'd go module + include, but your call.